var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -332,6 +379,34 @@ def setup_build_environment(self, env):
             lapack = 'lapack'
 
         env.set('NPY_LAPACK_ORDER', lapack)
+        # https://github.com/scipy/scipy/issues/9080
+        env.set('F90', spack_fc)
+
+    def fortran_args(self, spec, prefix):
+        args = []
+        if spec.satisfies('%fj'):
+            args.extend(['config_fc', '--fcompiler=fj'])
+        elif (spec.satisfies('platform=linux') and
+              spec.target.family == 'ppc64le' and
+              "xlf" in env['SPACK_F77']):
+            args.extend(['config_fc', '--fcompiler=ibm'])
+        return args
+
+    def build(self, spec, prefix):
+        """Install everything from build directory."""
+        args = (self.fortran_args(spec, prefix)
```

Comment:
I've been meaning to add some easy way to inject flags after `setup.py` but before `build/install`. If you want you can add a `setup_args` function to the base class in this PR, or I can add it at a later date. It should be easier to do this, and this isn't the first package that hacks this in.